experimental/perf/scripts/setup_helper.py

```diff
@@ -41,28 +45,44 @@ def get_logger(name: str, log_file: str = "perf.log") -> logging.Logger:
     return logger
 
 
-def generate_random_dict(num_fields: int, field_size: int) -> dict:
-    """Generates a JSON-like dict with the specified number of fields and field sizes.
+def get_payload(data_type: str, metric_count: int, metric_key_size) -> dict:
+    """Generates a payload based on the data type and parameters.
 
     Args:
-        num_fields (int): The number of key-value pairs (fields) in the JSON.
-        field_size (int): The size (in characters) of the field values.
+        data_type (str): scalar | audio - the wandb data type.
+        metric_count (int): The number of metrics.
+        metric_key_size (int): The size (in characters) of the metric.
 
     Returns:
-        str: A JSON-like string with the specified structure.
+        dict: payload for logging in the performance testing.
     """
+    if data_type == "audio":
+        duration = 5  # make a 5s long audio
+        sample_rate = 44100
+        frequency = 440
+
+        t = np.linspace(0, duration, int(sample_rate * duration), endpoint=False)
+        audio_data = np.sin(2 * np.pi * frequency * t)
+        audio_obj = wandb.Audio(audio_data, sample_rate=sample_rate)
+        return {get_random_key(metric_key_size): audio_obj for _ in range(metric_count)}
 
-    def random_key():
-        # Generate a random key with a length "field_size"  characters
-        return "".join(
-            random.choices(string.ascii_letters + string.digits + "_", k=field_size)
-        )
+    elif data_type == "scalar":
+        return {
+            get_random_key(metric_key_size): random.randint(1, 10**6)
+            for _ in range(metric_count)
+        }
 
-    # Generate the specified number of fields
-    return {random_key(): random.randint(1, 10**6) for _ in range(num_fields)}
+    return None
```

Comment:
you say that the return type is dict, but here you return None, either change the type def or maybe you can return an empty dict

Reply:
fixed